### PR TITLE
jsPromisesPracticeDOM: add tests

### PR DESCRIPTION
**Pull request by brainwashers**

### CODE REVIEW

File: cypress/integration/jsPromisesPracticeDOM.spec.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    cy.get('[data-qa="notification"]').contains(promise);
+  });
+
+const fPromRes = 'First promise was resolved';
+const fPromRej = 'First promise was rejected';
+const sPromRes = 'Second promise was resolved';
+const thPromRes = 'Third promise was resolved';
+
+describe('promises in DOM', () => {
+  beforeEach('open site', () => {
```
Comment (iBrianWarner) [inline cypress/integration/jsPromisesPracticeDOM.spec.js:20]:
```suggestion
describe('Promises in DOM', () => {
  beforeEach(() => {
```

**Review by iBrianWarner — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
+
 
 ### Task: Promises practice
 
 In this task, you should create 3 promises.
-1. The first promise should be resolved on a `click` in `document` or rejected in 5 seconds if not clicked.
-2. The second promise should be resolved only on `left` or `right` click on `document`. `middle` button should be ignored.
-3. The third promise should be resolved only if both `left` and `right` clicks happened.
+1. The first promise should be resolved on a `click` in `document` or rejected in 3 seconds if not clicked. Message: `First promise was resolved`, `First promise was rejected`
+2. The second promise should be resolved only on `left` or `right` click on `document`. `middle` button should be ignored. Message: `Second promise was resolved`.
```
Comment (roman-mate) [inline README.md:13]: What is middle button? :)

File: cypress/integration/jsPromisesPracticeDOM.spec.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+Cypress.Commands.add('checkPromise',
+  (promise) => {
+    cy.get('[data-qa="notification"]').contains(promise);
+  });
+
+const fPromRes = 'First promise was resolved';
+const fPromRej = 'First promise was rejected';
+const sPromRes = 'Second promise was resolved';
+const thPromRes = 'Third promise was resolved';
```
Comment (roman-mate) [inline cypress/integration/jsPromisesPracticeDOM.spec.js:17]: these names are too short. Don't shorten words to one character. 

firstResolvedMsg
firstRejectedMsg
...

File: cypress/integration/jsPromisesPracticeDOM.spec.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+Cypress.Commands.add('clickButton',
+  (mouseButton) => {
+    cy.get('body').trigger('mousedown', mouseButton)
+      .trigger('mouseup');
+  });
+
+Cypress.Commands.add('checkPromise',
```
Comment (roman-mate) [inline cypress/integration/jsPromisesPracticeDOM.spec.js:9]: checkPromise - need a better name. what exactly are we checking?

File: cypress/integration/jsPromisesPracticeDOM.spec.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+Cypress.Commands.add('clickButton',
+  (mouseButton) => {
+    cy.get('body').trigger('mousedown', mouseButton)
+      .trigger('mouseup');
```
Comment (roman-mate) [inline cypress/integration/jsPromisesPracticeDOM.spec.js:6]: does this fit in one line?

File: cypress/integration/jsPromisesPracticeDOM.spec.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  });
+
+Cypress.Commands.add('checkPromise',
+  (promise) => {
+    cy.get('[data-qa="notification"]').contains(promise);
+  });
```
Comment (roman-mate) [inline cypress/integration/jsPromisesPracticeDOM.spec.js:12]: are parentheses {} needed?

File: cypress/integration/jsPromisesPracticeDOM.spec.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    cy.checkPromise(fPromRes);
+  });
+
+  it('should ignore middle button click for the second promise', function() {
+    cy.clickButton({ button: 1 });
+    cy.clickButton({ button: 1 });
+    cy.checkPromise(sPromRes).should('not.exist');
+  });
+
+  it('should reject first promise ', function() {
```
Comment (roman-mate) [inline cypress/integration/jsPromisesPracticeDOM.spec.js:49]:
```suggestion
  it('should reject first promise after 3 seconds of inactivity', function() {
```

File: cypress/integration/jsPromisesPracticeDOM.spec.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+    // NOTE: waiting for reject
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(3000);
+    cy.checkPromise(fPromRej);
+  });
+
+  it('should resolve second promise right click', function() {
```
Comment (roman-mate) [inline cypress/integration/jsPromisesPracticeDOM.spec.js:56]:
```suggestion
  it('should resolve second promise on right click', function() {
```

File: cypress/integration/jsPromisesPracticeDOM.spec.js [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+    cy.clickButton({ button: 2 });
+    cy.checkPromise(sPromRes);
+  });
+
+  it('should resolve second promise left click', function() {
```
Comment (roman-mate) [inline cypress/integration/jsPromisesPracticeDOM.spec.js:61]:
```suggestion
  it('should resolve second promise on left click', function() {
```

File: cypress/integration/jsPromisesPracticeDOM.spec.js [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+    cy.checkPromise(thPromRes);
+  });
+
+  it('should resolve all promises', function() {
+    cy.clickButton({ button: 2 });
+    cy.clickButton({ button: 0 });
+    cy.checkPromise(fPromRes);
+    cy.checkPromise(sPromRes);
+    cy.checkPromise(thPromRes);
+  });
```
Comment (roman-mate) [inline cypress/integration/jsPromisesPracticeDOM.spec.js:78]: does this test case cover the previous test case?

File: README.md (hunk 2 above)
Comment (brainwashers) [reply README.md:13]: the mouse wheel

File: cypress/integration/jsPromisesPracticeDOM.spec.js (hunk 6 above)
Comment (brainwashers) [reply cypress/integration/jsPromisesPracticeDOM.spec.js:12]: According to documentation yes

File: cypress/integration/jsPromisesPracticeDOM.spec.js (hunk 10 above)
Comment (brainwashers) [reply cypress/integration/jsPromisesPracticeDOM.spec.js:78]: the previous one covers the case for the third promise only. this test case covers promises altogether

File: cypress/integration/jsPromisesPracticeDOM.spec.js (hunk 5 above)
Comment (brainwashers) [reply cypress/integration/jsPromisesPracticeDOM.spec.js:6]: yes

File: cypress/integration/jsPromisesPracticeDOM.spec.js (hunk 4 above)
Comment (brainwashers) [reply cypress/integration/jsPromisesPracticeDOM.spec.js:9]: Maybe 'findPromise' or 'checkPromiseExists'

File: cypress/integration/jsPromisesPracticeDOM.spec.js (hunk 5 above)
Comment (brainwashers) [reply cypress/integration/jsPromisesPracticeDOM.spec.js:6]: Changing

**Review by roman-mate — APPROVED** [review]

**Pull request merged**